### PR TITLE
[SSK-2764] fix: patch cluster status

### PR DESCRIPTION
**Pull request by apoorvprecisely**

### CODE REVIEW

**Comment by keshareenv** [comment]
> _No description provided._

is this a complete port of https://issues.apache.org/jira/browse/SOLR-10720, i see some changes to `OverseerCollectionMessageHandler` as well in that patch

**Comment by apoorvprecisely** [comment]
adding it, paths have changed

**Review by keshareenv — APPROVED** [review]
lgtm, please add git secrets before merging, then we can verify actions build

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        
+      - name: Setup Maven repository
+        id: configure-maven-settings
+        run: cp $GITHUB_WORKSPACE/.maven.settings.xml $HOME/.m2/settings.xml
+
+      - name: Build binaries
+        if: github.event_name == 'release'
+        env:
+          TAG: ${{ github.event.release.tag_name }}
+          BUILD_TAG: ${{ steps.extract_tag.outputs.BUILD_TAG }}
```
Comment (keshareenv) [inline .github/workflows/build.yaml:67]: step extract_tag missing

**Review by keshareenv — CHANGES_REQUESTED** [review]

**Comment by uknth** [comment]
Split this code in 2 PRs. 
1. Migration from Travis to Github Actions
2. Code Change with Test Cases

**Pull request closed without merging**